python/test/test_api.py

```diff
@@ -499,3 +645,54 @@ def test_parameters(self, server_options):
                 output_memory_type="cpu",
                 raise_on_error=True,
             )
+
+
+class TestMemoryLeak:
+    @staticmethod
+    @contextmanager
+    def _leak_detector(min_mem_inc, max_mem_inc):
+        gc.collect()
+        tracemalloc.start()
+        current, peak = tracemalloc.get_traced_memory()
+        yield
+        new_current, new_peak = tracemalloc.get_traced_memory()
+        mem_inc = new_current - current
+        assert mem_inc >= min_mem_inc
+        assert mem_inc < max_mem_inc
+        tracemalloc.stop()
+
+    # max_infer_repeat / min_infer_repeat >> max_mem_inc / min_mem_inc, so the memory
+    # increase is not proportional to the increase in inference requests.
```

Comment:
If there is no per-request/response memory leaks, shouldn't sending ~200 requests and ~8000 requests with the same concurrency/batching see roughly the same amount of memory usage before/after?

Do you have more detailed results/findings from local experiments that led you to these numbers and methodology?

Reply:
I found the memory usage increment will start higher and converge to around 40KB locally on my machine as the test progresses, regardless of the requests count, i.e.
1. requests = 200; memory increment = 71193
2. requests = 500; memory increment = 60405
3. requests = 1000; memory increment = 40133
4. requests = 8000; memory increment = 38519

and in the opposite order:
1. requests = 8000; memory increment = 83592
2. requests = 1000; memory increment = 47596
3. requests = 500; memory increment = 41406
4. requests = 200; memory increment = 40853

requests = 1000 repeat 6x:
1. memory increment = 84436
2. memory increment = 41967
3. memory increment = 40166
4. memory increment = 45353
5. memory increment = 41281
6. memory increment = 34740